docs/user_guide.md

```diff
@@ -0,0 +1,37 @@
+@mainpage
```

Comment:
How is this user_guide.mdfile used?  Is it needed for doxygen?

Reply:
Yes, it is needed. It forms the main page of the documentation. On the main page we can organize our lists of functions and our text about the code. For example, if you look here: https://www.unidata.ucar.edu/software/netcdf/docs/index.html you will see the netcdf-c library main page of documentation, with an introduction and then links to different parts of the documentation.

The .md file will contain what you now have in .rst files. Then we will produce the .rst files from the doxygen documentation, so we only have the documentation in one place.

But before we do that, we have to get the doxygen documentation working. ;-)

Reply:
Does that mean that the full Users Guide and technical inline code documentation from doxygen will live together in the same document?  Or just that there will eventually be one set of  UG rst file that we can continue to serve up separate from technical doxygen info?  Does this make sense?  I appreciate your time in helping me understand how these all work together!

Reply:
The first round of effort will get doxygen into your code and the .rst documentation will co-exist. You can evaluate how doxygen works and what the output is. The .rst documentation will be completely separate.

After that is complete, we can make the doxygen build create the .rst files as well. In that case, the doxygen documentation serves all documentation needs. This is what we will be doing on the other NCEPLIBS projects - but I have not tried it yet. ;-)

Reply:
Ok, this approach sounds good to me.  We have a good amount of updates to current rst files that will be coming into develop branch from public-v2 branch after SRW release next month, so I also just want to try to understand if that's going to require manual updating of this new user_guide.md while in develop branch as well.

Reply:
Good to know. I will not take any further material from the rst files until after the update from public-v2.

My first task, which will take some time, is to convert code comments to doxygen format. So I will not need to look at the .rst files until after that is complete.

Reply:
Sounds good, as with some of the other tasks you propose, we have a number of big releases and merges in the pipeline so it is probably best to wait until these are complete before making too many changes in develop.  As of right now I expect public-v2 to be merged in early December.

Reply:
OK, so you don't want me to edit code files now?

@WenMeng-NOAA can I proceed to edit some code files, or should I wait until December?

Reply:
Source code should be largely unaffected by merge of public-v2, it's mostly documentation and build utilities updates, so that merge will take on develop version of source code for the most part.  We needed the refactor branch merged first before you started with source code, which is now complete.  From community/public branch perspective it's ok with me to proceed with source code edits, but @WenMeng-NOAA should approve first.

Reply:
@fossell Your plan sounds good to me. Will there be a large changes in source for public-v2 merge? If not, it would be fine for me that @edwardhartnett proceed code editing for doxygen templates.

Reply:
@fossell Thanks for commenting. I would be fine for me to approve this PR. @edwardhartnett Will the source code editing be a separate PR or this PR?

Reply:
The source code editing will be separate PRs.

This PR just puts the build system in place. This is necessary but won't give us much until we modify the comments in the code files.

So please merge this PR, and I will put in a new PR with some code changes, so you can see how that will look.